### PR TITLE
Filter out concluded groups from calendar filter

### PR DESCRIPTION
**Pull request by vargaat**
refs: [MBL-17646](https://instructure.atlassian.net/browse/MBL-17646)
affects: Student
release note: Fixed calendar trying to load events for concluded groups.

test plan: See ticket

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] Tested on phone
- [ ] Tested on tablet

[MBL-17646]: https://instructure.atlassian.net/browse/MBL-17646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
#### Release Note: 
Fixed calendar trying to load events for concluded groups.

#### Affected Apps: Student

[MBL-17646](https://instructure.atlassian.net/browse/MBL-17646)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.61% | 91.6% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9b56591205e5a1bc96595aa0165024eede9aca7c
</p>

File: Core/Core/Planner/CalendarFilter/Model/UseCase/GetCalendarFilters.swift [hunk 1]
```diff
@@ -74,10 +74,15 @@ class GetCalendarFilters: UseCase {
...
             .CombineLatest(coursesFetch, groupsFetch)
-            .map {
-                APIResponse(
-                    courses: $0.0,
-                    groups: $0.1.body
+            .map { (courses, groups) in
+                let validCourseIDs = courses.map { $0.id }
+                /// If a course's end date has passed and "Restrict students from viewing course after course end date" is checked
+                /// then fetching events for a group in this course will give 403 unauthorized.
+                /// To be on the safe side we drop all courses without a valid course.
```
Comment (szabinst) [inline Core/Core/Planner/CalendarFilter/Model/UseCase/GetCalendarFilters.swift:81]: I guess it should be "drop all groups without"

**Review by szabinst — APPROVED** [review]
QA + 1

**Review by rh12 — DISMISSED** [review]
QA + 1

File: Core/Core/Planner/CalendarFilter/Model/UseCase/GetCalendarFilters.swift (hunk 1 above)
Comment (vargaat) [reply Core/Core/Planner/CalendarFilter/Model/UseCase/GetCalendarFilters.swift:81]: Agree, renamed.

**Review by rh12 — APPROVED** [review]

**Pull request merged**